spotbugs/src/main/java/edu/umd/cs/findbugs/detect/InefficientReplaceAll.java

```diff
@@ -0,0 +1,128 @@
+import edu.umd.cs.findbugs.ba.ClassContext;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+import edu.umd.cs.findbugs.classfile.MethodDescriptor;
+
+/**
+ * Find occurrences of replaceAll(String regex, String replacement) without any
+ * special regex characters.
+ *
+ * @author Lasse Lindqvist
+ */
+public class InefficientReplaceAll extends OpcodeStackDetector implements StatelessDetector {
+    private static final boolean DEBUG = SystemProperties.getBoolean("ira.debug");
+
+    private static final Logger LOGGER = Logger.getLogger(InefficientReplaceAll.class.getName());
+
+    private static final List<MethodDescriptor> methods = Collections.singletonList(
+            new MethodDescriptor("", "replaceAll", "(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;"));
+
+    private final BugAccumulator bugAccumulator;
+
+    private static final Pattern REGEX_CHARS_PATTERN = Pattern.compile("[\\-\\.\\\\\\[\\]\\{\\}\\(\\)\\<\\>\\*\\+\\=\\?\\^\\$\\|]");
+
+    public InefficientReplaceAll(BugReporter bugReporter) {
+        this.bugAccumulator = new BugAccumulator(bugReporter);
+    }
+
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        if (hasInterestingMethod(classContext.getJavaClass().getConstantPool(), methods)) {
+            classContext.getJavaClass().accept(this);
+        }
+    }
+
+    @Override
+    public void visit(Method obj) {
+        if (DEBUG) {
+            LOGGER.log(Level.INFO, "------------------- Analyzing " + obj.getName() + " ----------------");
+        }
+        super.visit(obj);
+    }
+
+    @Override
+    public void visit(Code obj) {
+        super.visit(obj);
+        bugAccumulator.reportAccumulatedBugs();
+
+    }
+
+    @Override
+    public void sawOpcode(int seen) {
+        if (DEBUG) {
+            LOGGER.log(Level.INFO, "Opcode: " + Const.getOpcodeName(seen));
+        }
+        if (((seen == Const.INVOKEVIRTUAL) || (seen == Const.INVOKEINTERFACE))
+                && ("replaceAll".equals(getNameConstantOperand()))
+                && ("(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;".equals(getSigConstantOperand()))
+                && hasConstantArguments()) {
+            Optional<String> firstArgument = getFirstArgument();
+            if (firstArgument.isPresent()) {
+                if (!REGEX_CHARS_PATTERN.matcher(firstArgument.get()).find()) {
+                    bugAccumulator.accumulateBug(
+                            new BugInstance(this, "IRA_INEFFICIENT_REPLACEALL", LOW_PRIORITY).addClassAndMethod(this),
+                            this);
+                }
+            }
+        }
+    }
+
+    /**
+     * @return first argument of the called method if it's a constant
+     */
+    private Optional<String> getFirstArgument() {
+        Object value = getStack().getStackItem(getNumberArguments(getMethodDescriptorOperand().getSignature()) - 1)
+                .getConstant();
+        if (value == null) {
```

Comment:
Simpler:

    return Optional.ofNullable(Objects.toString(value, null));

Reply:
I will change that.